tests/unit/reaction/conservation_rules/test_additive_qn.py

```diff
@@ -8,6 +8,4 @@
     [(([0], [-1, 1]), True), (([0], [1, 1]), False)],
 )
 def test_charge_conservation(graph_input, expected_value):
-    rule = ChargeConservation()
-
-    assert rule(*graph_input) == expected_value
+    assert ChargeConservation()(*graph_input) == expected_value
```

Comment:
See question on why the rules are still classes.

Reply:
The rules created with the additive quantum number decorator have the additional difficulty to achieve the pickling support. With classes this much easier to sidestep

Reply:
Ah yes, the pickling was the problem